hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -102,7 +102,7 @@ class DefaultSource extends RelationProvider
       )
     } else {
       Map()
-    }) ++ DataSourceOptionsHelper.parametersWithReadDefaults(sqlContext.getAllConfs.filter(k => k._1.startsWith("hoodie.")) ++ optParams)
+    }) ++ DataSourceOptionsHelper.parametersWithReadDefaults(extractHoodieConfig(sqlContext.getAllConfs) ++ optParams)
```

Comment:
Please add some tests, such as running the job of `procedure`s, `command`s, and queries.